template-for-repository/proofs/Makefile.common

```diff
@@ -300,54 +273,137 @@ CBMC_REMOVE_FUNCTION_BODY := $(patsubst %,--remove-function-body %, $(REMOVE_FUN
 
 # Compile project sources
 $(PROJECT_GOTO)1.goto: $(PROJECT_SOURCES)
-	mkdir -p $(dir $@)
-	mkdir -p $(dir $(call LOG_FROM_ENTRY,$@))
-	$(call DO_GOTO_CC,--export-file-local-symbols $(COMPILE_FLAGS) $(INCLUDES) $(DEFINES),$^,$@)
+	$(LITANI) add-job \
+	  --command \
+	    '$(GOTO_CC) $(CBMC_VERBOSITY) --export-file-local-symbols $(COMPILE_FLAGS) $(INCLUDES) $(DEFINES) $^ -o $@' \
+	  --inputs $^ \
+	  --outputs $@ $(call LOG_FROM_ENTRY,$@) \
+	  --pipeline-name $(HARNESS_ENTRY) \
+	  --ci-stage build \
+	  --interleave-stdout-stderr \
+	  --stdout-file $(call LOG_FROM_ENTRY,$@) \
+	  --description "$(HARNESS_ENTRY): building project binary"
 
 # Compile proof sources
 $(PROOF_GOTO)1.goto: $(PROOF_SOURCES)
-	mkdir -p $(dir $@)
-	mkdir -p $(dir $(call LOG_FROM_ENTRY,$@))
-	$(call DO_GOTO_CC,--export-file-local-symbols $(COMPILE_FLAGS) $(INCLUDES) $(DEFINES),$^,$@)
+	$(LITANI) add-job \
+	  --command \
+	    '$(GOTO_CC) $(CBMC_VERBOSITY) --export-file-local-symbols $(COMPILE_FLAGS) $(INCLUDES) $(DEFINES) $^ -o $@' \
+	  --inputs $^ \
+	  --outputs $@ $(call LOG_FROM_ENTRY,$@) \
+	  --pipeline-name $(HARNESS_ENTRY) \
+	  --ci-stage build \
+	  --interleave-stdout-stderr \
+	  --stdout-file $(call LOG_FROM_ENTRY,$@) \
```

Comment:
This is more of a litani question, but I'm a bit surprised that outputs includes the log.

Reply:
There are two reasons for this:
* Litani copies all output files to the report directory, so that they can be viewed in a web browser.
* Jobs that depend on an output of this job will not be invoked until this job terminates successfully. This doesn't matter in this particular case since nobody depends on this log. Further down, you'll notice that CBMC Viewer does depend on the logs of CBMC.

The build graph really is a graph, not an upside-down tree. Multiple outputs are fine, ninja will do the right thing.